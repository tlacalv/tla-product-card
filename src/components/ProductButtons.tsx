import styles from "../styles/styles.module.css";
import React, { CSSProperties, useCallback, useContext } from "react";
import { ProductContext } from "./ProductCard";

export interface Props {
  className?: string;
  style?: CSSProperties;
}

export const ProductButtons = ({ className, style }: Props) => {
  const { increaseBy, counter, maxCount } = useContext(ProductContext);

  const isMaxReached = useCallback(
    () => !!maxCount && maxCount===counter,
    [counter, maxCount]
  )

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button className={styles.buttonMinus} onClick={() => increaseBy(-1)}>
        -
      </button>
      <div className={styles.countLabel}>{counter}</div>
      <button disabled={isMaxReached()} className={`${styles.buttonAdd} ${isMaxReached() && styles.disabled}`} onClick={() => increaseBy(1)}>
        +
      </button>
    </div>
  );
};
