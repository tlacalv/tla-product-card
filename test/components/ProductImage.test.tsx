import React from 'react'
import renderer from 'react-test-renderer'
import {ProductImage, ProductCard} from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
  test('must show the component with customized image', () => {
    const wrapper = renderer.create(<ProductImage img ="https://hola.jpg"/>)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  test('must show the component with the product image', () => {
    const wrapper = renderer.create(<ProductCard product={product2}>
      {
        () => (
          <ProductImage />
        )
      }
    </ProductCard>)

    expect(wrapper.toJSON()).toMatchSnapshot();
  })
})