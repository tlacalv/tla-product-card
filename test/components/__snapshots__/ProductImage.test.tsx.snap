// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage must show the component with customized image 1`] = `
<img
  alt="Product Image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage must show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
